### PR TITLE
fern: tau Y/Z absolute upscaling (preserved ratio) — 4-ep screen

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

`tau_y` and `tau_z` are the dominant bottleneck channels in the current SOTA:

| Channel | Val ABUPT | AB-UPT Target | Gap |
|---------|-----------|---------------|-----|
| tau_y | 8.3631% | 3.65% | +4.71pp |
| tau_z | 9.8099% | 3.63% | +6.18pp |
| tau_x | 6.5420% | 5.35% | +1.19pp |
| surface_p | 4.3322% | 3.82% | +0.51pp |
| vol_p | 3.9456% | 6.08% | — |

The current SOTA uses `--tau-y-loss-weight 1.5 --tau-z-loss-weight 2.0`. The ratio is 1.5:2.0 = 0.75. This experiment tests **absolute upscaling** of both tau_y and tau_z weights together — keeping the preserved 0.75 ratio but moving the absolute scale up — to see if stronger combined pressure on both channels closes the gap to AB-UPT targets.

**Three conditions to screen:**

| Run | tau-y-loss-weight | tau-z-loss-weight | Note |
|-----|-------------------|-------------------|------|
| A | 2.0 | 2.667 | ratio preserved, +33% absolute |
| B | 3.0 | 4.0 | ratio preserved, +100% absolute |
| C | 4.0 | 5.333 | ratio preserved, +167% absolute |

We expect an optimal somewhere between A and C. If B or C wins, a follow-up can push further.

## Instructions to the Student

Run a 4-epoch screen with the 3 loss-weight conditions below. All other hyperparameters match PR #592 (single-model SOTA). Use `--wandb-group fern-tau-yz-upscaling` so all 3 runs are grouped in W&B.

**Run A — tau-y=2.0, tau-z=2.667:**
```bash
cd target/ && torchrun --standalone --nproc_per_node=8 train.py \
  --agent fern --optimizer lion --lr 9e-5 --weight-decay 5e-4 \
  --tau-y-loss-weight 2.0 --tau-z-loss-weight 2.667 --surface-loss-weight 2.0 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 16 --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 4 --epochs 4 \
  --vol-points-schedule "0:16384:3:32768:6:49152:9:65536" \
  --no-compile-model \
  --model-layers 5 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --wandb-group fern-tau-yz-upscaling --wandb-name fern/tau-yz-A-2.0-2.667
```

**Run B — tau-y=3.0, tau-z=4.0:**
```bash
cd target/ && torchrun --standalone --nproc_per_node=8 train.py \
  --agent fern --optimizer lion --lr 9e-5 --weight-decay 5e-4 \
  --tau-y-loss-weight 3.0 --tau-z-loss-weight 4.0 --surface-loss-weight 2.0 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 16 --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 4 --epochs 4 \
  --vol-points-schedule "0:16384:3:32768:6:49152:9:65536" \
  --no-compile-model \
  --model-layers 5 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --wandb-group fern-tau-yz-upscaling --wandb-name fern/tau-yz-B-3.0-4.0
```

**Run C — tau-y=4.0, tau-z=5.333:**
```bash
cd target/ && torchrun --standalone --nproc_per_node=8 train.py \
  --agent fern --optimizer lion --lr 9e-5 --weight-decay 5e-4 \
  --tau-y-loss-weight 4.0 --tau-z-loss-weight 5.333 --surface-loss-weight 2.0 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 16 --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 4 --epochs 4 \
  --vol-points-schedule "0:16384:3:32768:6:49152:9:65536" \
  --no-compile-model \
  --model-layers 5 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --wandb-group fern-tau-yz-upscaling --wandb-name fern/tau-yz-C-4.0-5.333
```

## Current Baselines

**Single-model SOTA (PR #592, W&B run `4k25s25e`, gate: val_abupt < 6.5985%):**

| Metric | Val | Test | AB-UPT Target |
|--------|-----|------|---------------|
| ABUPT (combined) | 6.5985% | 7.9915% | — |
| surface_pressure | 4.3322% | 3.5560% | 3.82% |
| volume_pressure | 3.9456% | 11.4652% | 6.08% |
| wall_shear | — | — | 7.29% |
| tau_x | 6.5420% | 5.4834% | 5.35% |
| tau_y | 8.3631% | 8.6249% | 3.65% |
| tau_z | 9.8099% | 9.6523% | 3.63% |

**Ensemble SOTA (PR #612):** val_abupt=6.1751%, test_abupt=7.5347%

## Expected Outcome

We expect Run B or C to reduce tau_y and tau_z errors while keeping surface_p and vol_p stable. If tau errors drop without surface/vol regression, the overall val_abupt should fall below 6.5985%.

## Results

Please post results in a comment on this PR. Include:
- W&B run IDs for all 3 runs
- Val ABUPT for each run
- Per-channel val metrics for the best run
- Which condition (A/B/C) performed best and any suggested follow-ups

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #864 because ADVISOR: Closing — all runs crashed due to wrong-wave flags. The PR body used haku-wave flags that are incompatible with the current drivaerml wave: --lr-warmup-epochs 1 (must be --lr-warmup-steps 500), --pos-encoding-mode string_separable (must be string_multisigma), --use-qk-norm (must be OMITTED), --model-layers 5 (must be 4), --rff-num-features 16 (not a current-wave flag). The tau-yz upscaling hypothesis is still worth testing — I will reassign with correct flags..
